sw/device/lib/dif/dif_entropy_src.c

```diff
@@ -48,6 +52,35 @@ static void set_config_register(const dif_entropy_src_t *entropy,
                       reg);
 }
 
+/**
+ * Sets the `entropy` source firmware override register with the settings
+ * derived from `config`.
+ */
+static dif_entropy_src_result_t fw_override_set(
+    const dif_entropy_src_t *entropy,
+    const dif_entropy_src_fw_override_config_t *config) {
+  if (config->buffer_threshold > kDifEntropySrcFifoMaxCapacity) {
+    return kDifEntropySrcInvalidFwOverrideBufferThreshold;
+  }
+
+  if (config->entropy_insert_enable && !config->enable) {
+    return kDifEntropySrcInvalidFwOverrideSettings;
+  }
+  mmio_region_write32(entropy->params.base_addr,
+                      ENTROPY_SRC_OBSERVE_FIFO_THRESH_REG_OFFSET,
+                      config->buffer_threshold);
+
+  uint32_t reg =
+      bitfield_field32_write(0, ENTROPY_SRC_FW_OV_CONTROL_FW_OV_MODE_FIELD,
+                             config->enable ? 0xa : 0x5);
+  reg = bitfield_field32_write(
+      reg, ENTROPY_SRC_FW_OV_CONTROL_FW_OV_ENTROPY_INSERT_FIELD,
+      config->entropy_insert_enable ? 0xa : 0x5);
+  mmio_region_write32(entropy->params.base_addr,
+                      ENTROPY_SRC_FW_OV_CONTROL_REG_OFFSET, reg);
+  return kDifEntropySrcOk;
```

Comment:
The return codes will need to change too upon rebase on #8412 (this will become `kDifOk`). Applies throughout as well.